### ISSUE TITLE
Bogus "Update needed" happening randomly, server restart needed

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#36850]
### Steps

1. Have OC 10.3.2 running for several days

### Expected result

Running continuously.

### Actual result

An unknown trigger is putting OC into "Update needed" state, despite no update being available and automated update is disabled.
<img width="252" alt="image" src="https://user-images.githubusercontent.com/277525/73336123-471c5c00-4271-11ea-9045-bc7b82fc5490.png">

The solution is to restart Apache and it works again.

### Version
openSUSE Leap 15.1
owncloud-files-10.3.2-lp151.2.1.noarch (from official RPMs)
php7-7.2.5-lp151.6.13.1.x86_64

### Env

If have APCu enabled.

### App list
```
Enabled:
  - activity: 2.5.1
  - calendar: 1.6.3
  - configreport: 0.2.0
  - contacts: 1.5.5
  - dav: 0.5.0
  - encryption: 1.4.0
  - federatedfilesharing: 0.5.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_mediaviewer: 1.0.0
  - files_pdfviewer: 0.11.0
  - files_sharing: 0.12.0
  - files_texteditor: 2.3.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firstrunwizard: 1.2.0
  - market: 0.5.0
Disabled:
  - comments
  - external
  - federation
  - files_videoplayer
  - gallery
  - notifications
  - provisioning_api
  - systemtags
  - updatenotification
  - user_external
  - xmas
```

### config.php
```
<?php
$CONFIG = array (
  'datadirectory' => '/var/lib/owncloud/data',
  'cache_path' => '/var/cache/owncloud/cache',
  'previews_path' => '/var/cache/owncloud/thumbnails',
  'dav.chunk_base_dir' => '/var/cache/owncloud/uploads',
  'dbtype' => 'mysql',
  'version' => '10.3.2.2',
  'installed' => true,
  'lastupdatedat' => '1335993186.8567',
  'instanceid' => '*************',
  'maintenance' => false,
  'theme' => '',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => '*****************',
  'loglevel' => '0',
  'logfile' => '/var/log/owncloud/owncloud.log',
  'updatechecker' => false,
  'trashbin_purge_limit' => 25,
  'trusted_domains' => 
  array (
    0 => 'owncloud.vincentpetry.net',
  ),
  'secret' => '********************************************************************',
  'appcodechecker' => false,
  'filesystem_check_changes' => 0,
  'appstore.experimental.enabled' => true,
  'trashbin_retention_obligation' => 'auto',
  'overwrite.cli.url' => 'https://owncloud.vincentpetry.net/',
  'htaccess.RewriteBase' => '/',
  'singleuser' => false,
  'upgrade.disable-web' => true,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/srv/www/htdocs/owncloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/srv/www/htdocs/owncloud/apps-external',
      'url' => '/apps-external',
      'writable' => true,
    ),
  ),
);
```

So far this is only happening every few days, randomly, so I can't correlate this to anything.
I'll continue observing and reporting.

**Comment by cortho** [comment]
At a first glance I'd say this comes from the files_mediaviewer app where v 1.0.1 is available on marketplace. Could it possible the upgrade notification comes from that app?

**Comment by PVince81** [comment]
It might be triggered by some app update availability but I didn't see any related to the time frame where the bug happened. I also don't remember exactly where/when such check is done.

I remember a cron job in the market app that is only here to send notifications (through notifications app) to the admin in case of updates, but I don't think it would affect the regular update code.

As far as I remember that page is displayed either when the version stored in version.php differs from the one in config.php (which wasn't the case for my setup), or there is app code on disk which version is higher than what is stored in oc_appconfig. Since I did not touch my setup and disabled automatic updates, the latter case cannot happen either.

I wonder if it's some kind of APCu bug where maybe sometimes it would use an older version of config.php... It's already a bug that config.php is cached at all. I noticed this when sometimes the maintenance mode is stuck. The solution was always restarting Apache.
